lib/Retcon/Store/PostgreSQL.hs

```diff
@@ -43,18 +45,54 @@ fromSuccess _           = error "fromSuccess: Cannot unwrap not-a-success."
 isSuccess (Success _)   = True
 isSuccess   _           = False
 
+-- | Update ekg values for everything
+updateEkg :: Connection -> IO ()
+updateEkg conn = do
+    updateIKCounts      conn
+    updateFKCounts      conn
+    updateConflicts     conn
+    updateNotifications conn
+
+-- | Update ekg values for internal keys
+updateIKCounts :: Connection -> IO ()
+updateIKCounts conn = do
+    res <- query_ conn "SELECT COUNT(DISTINCT id), entity FROM retcon GROUP BY entity"
+    forM_ res $ uncurry setEntityKeys . second EntityName
+
+-- | Update ekg values for foreign keys
+updateFKCounts :: Connection -> IO ()
+updateFKCounts conn = do
+    res <- query_ conn "SELECT COUNT(DISTINCT id), entity, source FROM retcon_fk GROUP BY source, entity"
+    forM_ res $ \(c, e, s) -> setSourceKeys c (EntityName e) (SourceName s)
+
+-- | Update ekg values for conflicts
+updateConflicts :: Connection -> IO ()
+updateConflicts conn = do
+    res <- query_ conn "SELECT COUNT(DISTINCT diff_id), entity FROM retcon_diff WHERE is_conflict GROUP BY entity"
+    forM_ res $ \(c, e) -> setConflicts c (EntityName e)
+
+-- | Update ekg values for notifications
+updateNotifications :: Connection -> IO ()
+updateNotifications conn = do
+    res <- query_ conn "SELECT COUNT(DISTINCT id), entity FROM retcon_notifications GROUP BY entity"
+    let total = sum $ map fst res
+    forM_ res $ \(c, e) -> setEntityNotifications c (EntityName e)
+    setServerNotifications total
+
 -- | Persistent PostgreSQL-backed data storage.
 instance Store PGStore where
   newtype StoreOpts PGStore = PGOpts { connstr :: ByteString }
 
   initBackend (connstr -> str) = do
       conn <- connectPostgreSQL str
+      updateEkg conn
```

Comment:
It's not obvious to me when this is called - from the name, it seems like either once or once per operation? Either way, could you add a clarifying comment here?